src/stackdriver_agent.c

```diff
@@ -146,39 +156,52 @@ static int sagt_read(user_data_t *user_data)
   // Stats for API requests. The corresponding uc_meta_data_set calls are in
   // write_gcm.c.
   {
-    uint64_t value;
-    if (uc_meta_data_get_unsigned_int(
-            &vl, SAGT_API_REQUESTS_SUCCESS, &value) == 0) {
-      sagt_submit_derive("api_request_count", "success",
-                         now, interval, value, NULL);
-    }
-    if (uc_meta_data_get_unsigned_int(
-            &vl, SAGT_API_REQUESTS_CONNECTIVITY_FAILURES, &value) == 0) {
-      sagt_submit_derive("api_request_count", "connectivity_failures",
-                         now, interval, value, NULL);
-    }
-    if (uc_meta_data_get_unsigned_int(
-            &vl, SAGT_API_REQUESTS_ERRORS, &value) == 0) {
-      sagt_submit_derive("api_request_count", "errors",
-                         now, interval, value, NULL);
+    meta_data_t *md = meta_data_create();
+    if (meta_data_add_string(
+            md, "stackdriver_metric_type",
+            AGENT_PREFIX "/api_request_count") == 0) {
+      uint64_t value;
+      if (uc_meta_data_get_unsigned_int(
+              &vl, SAGT_API_REQUESTS_SUCCESS, &value) == 0 &&
+          meta_data_add_string(md, "label:state", "success") == 0) {
+        sagt_submit_derive("", NULL, now, interval, value, md);
+      }
+      if (uc_meta_data_get_unsigned_int(
+              &vl, SAGT_API_REQUESTS_CONNECTIVITY_FAILURES, &value) == 0 &&
+          meta_data_add_string(
+              md, "label:state", "connectivity_failures") == 0) {
+        sagt_submit_derive("", NULL, now, interval, value, md);
+      }
+      if (uc_meta_data_get_unsigned_int(
+              &vl, SAGT_API_REQUESTS_ERRORS, &value) == 0 &&
+          meta_data_add_string(md, "label:state", "errors") == 0) {
+        sagt_submit_derive("", NULL, now, interval, value, md);
+      }
     }
+    meta_data_destroy(md);
   }
 
   // Cloud Monarch-related stats. The corresponding uc_meta_data_set calls are in
   // match_throttle_metadata_keys.c.
   {
+    meta_data_t *md = meta_data_create();
```

Comment:
Nominally, this would be more readable just before the conditional, but up to you.

Reply:
Since you prompted it, I tried both. I find it slightly preferable to have metadata surround the entire block, so unless there is a significant reason to make the change, I'll leave it as-is.